async-query/src/main/java/org/opensearch/sql/spark/config/SparkExecutionEngineConfigSupplierImpl.java

```diff
@@ -6,20 +6,17 @@
 package org.opensearch.sql.spark.config;
 
 import static org.opensearch.sql.common.setting.Settings.Key.CLUSTER_NAME;
-import static org.opensearch.sql.common.setting.Settings.Key.SPARK_EXECUTION_ENGINE_CONFIG;
 
-import java.security.AccessController;
-import java.security.PrivilegedAction;
 import lombok.AllArgsConstructor;
-import org.apache.commons.lang3.StringUtils;
 import org.opensearch.cluster.ClusterName;
 import org.opensearch.sql.common.setting.Settings;
 import org.opensearch.sql.spark.asyncquery.model.AsyncQueryRequestContext;
 
 @AllArgsConstructor
 public class SparkExecutionEngineConfigSupplierImpl implements SparkExecutionEngineConfigSupplier {
 
-  private Settings settings;
+  private final Settings settings;
```

Comment:
in case we proposal setting loader, could we also unify setting loader and config supplier concept?

Reply:
The setting loader was very specific to the SparkExecutionEngineConfigClusterSetting so that we don't need to write duplicate code to verify privilege and load the setting from two place.